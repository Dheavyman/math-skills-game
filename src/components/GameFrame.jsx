import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Stars from './Stars';
import Button from './Button';
import Answer from './Answer';
import Numbers from './Numbers';
import DoneFrame from './DoneFrame';

const propTypes = {
  selectedNumbers: PropTypes.arrayOf(PropTypes.number).isRequired,
  usedNumbers: PropTypes.arrayOf(PropTypes.number).isRequired,
  redraws: PropTypes.number.isRequired,
  countdownRunning: PropTypes.bool.isRequired,
  isAnswerCorrect: PropTypes.bool,
  numberOfStars: PropTypes.number.isRequired,
  doneStatus: PropTypes.string,
  handleCheckAnswer: PropTypes.func.isRequired,
  handleAcceptAnswer: PropTypes.func.isRequired,
  handleRedraw: PropTypes.func.isRequired,
  handleUnselectNumber: PropTypes.func.isRequired,
  handleResetGame: PropTypes.func.isRequired,
  handleSelectNumber: PropTypes.func.isRequired,
};

const defaultProps = {
  isAnswerCorrect: null,
  doneStatus: null,
};

/**
 * Game frame component
 *
 * @param {object} props - Properties passed to component
 *
 * @returns {object} React element
 */
const GameFrame = (props) => {
  const {
    selectedNumbers,
    usedNumbers,
    redraws,
    countdownRunning,
    isAnswerCorrect,
    numberOfStars,
    doneStatus,
    handleCheckAnswer,
    handleAcceptAnswer,
    handleRedraw,
    handleUnselectNumber,
    handleResetGame,
    handleSelectNumber
  } = props;

  return (
    <Fragment>
      <div className="row mt-5">
        <Stars numberOfStars={numberOfStars} />
        <Button
          selectedNumbers={selectedNumbers}
          isAnswerCorrect={isAnswerCorrect}
          redraws={redraws}
          countdownRunning={countdownRunning}
          handleCheckAnswer={handleCheckAnswer}
          handleAcceptAnswer={handleAcceptAnswer}
          handleRedraw={handleRedraw}
        />
        <Answer
          selectedNumbers={selectedNumbers}
          handleUnselectNumber={handleUnselectNumber}
        />
      </div>
      <br />
      <br />
      {doneStatus
        ? (
          <DoneFrame
            doneStatus={doneStatus}
            handleResetGame={handleResetGame}
          />
        )
        : (
          <Numbers
            selectedNumbers={selectedNumbers}
            usedNumbers={usedNumbers}
            handleSelectNumber={handleSelectNumber}
          />
        )
      }
    </Fragment>
  );
};

GameFrame.propTypes = propTypes;
GameFrame.defaultProps = defaultProps;

export default GameFrame;
