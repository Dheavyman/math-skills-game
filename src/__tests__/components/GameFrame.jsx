import React from 'react';
import TestRenderer from 'react-test-renderer';

import GameFrame from '../../components/GameFrame';

describe('Game frame component', () => {
  let props = {
    selectedNumbers: [],
    usedNumbers: [],
    redraws: 5,
    countdownRunning: false,
    isAnswerCorrect: null,
    numberOfStars: 3,
    doneStatus: null,
    handleCheckAnswer: jest.fn(),
    handleAcceptAnswer: jest.fn(),
    handleRedraw: jest.fn(),
    handleUnselectNumber: jest.fn(),
    handleResetGame: jest.fn(),
    handleSelectNumber: jest.fn()
  };
  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<GameFrame {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component with doneStatus successfully', () => {
    props = {
      ...props,
      doneStatus: 'Done, Nice!'
    };
    const renderer = TestRenderer.create(<GameFrame {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
