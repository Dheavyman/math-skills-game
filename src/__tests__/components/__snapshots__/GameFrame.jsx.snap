// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game frame component should render component successfully 1`] = `
Array [
  <div
    className="row mt-5"
  >
    <div
      className="col-5"
    >
      <i
        className="fa fa-star"
      />
      <i
        className="fa fa-star"
      />
      <i
        className="fa fa-star"
      />
    </div>
    <div
      className="col-2 check-btn text-center"
    >
      <button
        className="btn btn-secondary btn-lg"
        disabled={true}
        onClick={[MockFunction]}
        type="button"
      >
        =
      </button>
      <br />
      <br />
      <button
        className="btn btn-warning btn-sm"
        disabled={true}
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fa fa-sync-alt fa-spin"
        />
          5
      </button>
    </div>
    <div
      className="col-5"
    >
      <div
        className="number"
      />
    </div>
  </div>,
  <br />,
  <br />,
  <div
    className="card border-primary text-center mt-5"
  >
    <div
      className="number"
    >
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        1
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        2
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        3
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        4
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        5
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        6
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        7
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        8
      </span>
      <span
        onClick={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        9
      </span>
    </div>
  </div>,
]
`;

exports[`Game frame component should render component with doneStatus successfully 1`] = `
Array [
  <div
    className="row mt-5"
  >
    <div
      className="col-5"
    >
      <i
        className="fa fa-star"
      />
      <i
        className="fa fa-star"
      />
      <i
        className="fa fa-star"
      />
    </div>
    <div
      className="col-2 check-btn text-center"
    >
      <button
        className="btn btn-secondary btn-lg"
        disabled={true}
        onClick={[MockFunction]}
        type="button"
      >
        =
      </button>
      <br />
      <br />
      <button
        className="btn btn-warning btn-sm"
        disabled={true}
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fa fa-sync-alt fa-spin"
        />
          5
      </button>
    </div>
    <div
      className="col-5"
    >
      <div
        className="number"
      />
    </div>
  </div>,
  <br />,
  <br />,
  <div
    className="text-center"
  >
    <h2>
      Done, Nice!
    </h2>
    <br />
    <button
      className="btn btn-info"
      onClick={[MockFunction]}
      type="button"
    >
      Play Again
    </button>
  </div>,
]
`;
